articles/api-auth/blacklists-vs-grants.md

```diff
@@ -0,0 +1,51 @@
+---
+description: This document covers blacklists vs. grants when it comes handling tokens.
+---
+
+# Blacklists and Client Grants
+
+Let's say that you're using a non-interactive [client](/clients) to represent your protected resource. You have a partner that calls your API, and at the end of your existing contract, you and your partner decide not to renew your partnership. As such, you now want to remove your partner's access to your API. The issue, however, is that you've given your partner an [access token](/tokens/access-token) that lasts for a month.
+
+* What can you do in this situation?
+* How might you configure your Auth0 environment to make such situations easier to handle in the future?
+
+In this article, we'll cover two methods for revoking access to the protected resource:
+
+* [Blacklisting the access token](#blacklists)
+* [Revoking the client grant](#client-grants)
+
+We will then compare the two methods and provide our recommendations.
+
+## Blacklists
+
+Let's say that you grant access to your API to anyone in possession of the appropriate access token. One method of revoking access to a user is to blacklist their token so that it can no longer be used.
+
+::: note
+Please see the Auth0 blog for an in-depth treatment on [Blacklisting JSON Web Token API Keys](https://auth0.com/blog/blacklist-json-web-token-api-keys/).
+:::
+
+Auth0-issued tokens are [JWTs](/jwt), so the identifier for each token is the JWT ID, or `jti`. With the `jti` in hand, you can make the appropriate `POST` call to the Management API's [blacklist a token endpoint](/api/management/v2#!/Blacklists/post_tokens). You'll need to provide the JWT's `aud` and `jti` claims. 
```

Comment:
@khorne3 Is this always the case? I just got an access_token from a sample SPA and when I decoded it I didn't have a `jti` claim. Should my request have something specific set?

Request:
```
https://auth0pnp.auth0.com/authorize?scope=read:timesheets%20openid%20profile&audience=https://api.abcinc.com/timesheets&response_type=id_token%20token&client_id=o6Hv1xl60a6UqSvQpbK9LDwT18kVJZci&redirect_uri=https://localhost/callback&nonce=123
```

Decoded access_token:
```
{
  "iss": "https://auth0pnp.auth0.com/",
  "sub": "google-oauth2|106091499222329415433",
  "aud": [
    "https://api.abcinc.com/timesheets",
    "https://auth0pnp.auth0.com/userinfo"
  ],
  "azp": "o6Hv1xl60a6UqSvQpbK9LDwT18kVJZci",
  "exp": 1497439793,
  "iat": 1497439733,
  "scope": "read:timesheets openid profile",
  "https://api.abcinc.com/email": "maria.paktiti@gmail.com"
}
```